src/languages/singlestoredb/singlestoredb.functions.ts

```diff
@@ -0,0 +1,302 @@
+    'GEOMETRY_POINT',
+    'GEOMETRY_WITHIN_DISTANCE',
+    'GEOMETRY_X',
+    'GEOMETRY_Y',
+    'GREATEST',
+    'GROUPING',
+    'GROUP_CONCAT',
+    'HEX',
+    'HIGHLIGHT',
+    'HOUR',
+    'ICU_VERSION',
+    'IF',
+    'IFNULL',
+    'INET_ATON',
+    'INET_NTOA',
+    'INET6_ATON',
+    'INET6_NTOA',
+    'INITCAP',
+    'INSERT',
+    'INSTR',
+    'INTERVAL',
+    'IS',
+    'IS NULL',
+    'JSON_AGG',
+    'JSON_ARRAY_CONTAINS_DOUBLE',
+    'JSON_ARRAY_CONTAINS_JSON',
+    'JSON_ARRAY_CONTAINS_STRING',
+    'JSON_ARRAY_PUSH_DOUBLE',
+    'JSON_ARRAY_PUSH_JSON',
+    'JSON_ARRAY_PUSH_STRING',
+    'JSON_DELETE_KEY',
+    'JSON_EXTRACT_DOUBLE',
+    'JSON_EXTRACT_JSON',
+    'JSON_EXTRACT_STRING',
+    'JSON_EXTRACT_BIGINT',
+    'JSON_GET_TYPE',
+    'JSON_LENGTH',
+    'JSON_SET_DOUBLE',
+    'JSON_SET_JSON',
+    'JSON_SET_STRING',
+    'JSON_SPLICE_DOUBLE',
+    'JSON_SPLICE_JSON',
+    'JSON_SPLICE_STRING',
+    'LAG',
+    'LAST_DAY',
+    'LAST_VALUE',
+    'LCASE',
+    'LEAD',
+    'LEAST',
+    'LEFT',
+    'LENGTH',
+    'LIKE',
+    'LN',
+    'LOCALTIME',
+    'LOCALTIMESTAMP',
+    'LOCATE',
+    'LOG',
+    'LOG10',
+    'LOG2',
+    'LPAD',
+    'LTRIM',
+    'MATCH',
+    'MAX',
+    'MD5',
+    'MEDIAN',
+    'MICROSECOND',
+    'MIN',
+    'MINUTE',
+    'MOD',
+    'MONTH',
+    'MONTHNAME',
+    'MONTHS_BETWEEN',
+    'NOT',
+    'NOW',
+    'NTH_VALUE',
+    'NTILE',
+    'NULLIF',
+    'OCTET_LENGTH',
+    'PERCENT_RANK',
+    'PERCENTILE_CONT',
+    'PERCENTILE_DISC',
+    'PI',
+    'PIVOT',
+    'POSITION',
+    'POW',
+    'POWER',
+    'QUARTER',
+    'QUOTE',
+    'RADIANS',
+    'RAND',
+    'RANK',
+    'REGEXP',
+    'REPEAT',
+    'REPLACE',
+    'REVERSE',
+    'RIGHT',
+    'RLIKE',
+    'ROUND',
+    'ROW_COUNT',
+    'ROW_NUMBER',
+    'RPAD',
+    'RTRIM',
+    'SCALAR',
+    'SCHEMA',
+    'SEC_TO_TIME',
+    'SHA1',
+    'SHA2',
+    'SIGMOID',
+    'SIGN',
+    'SIN',
+    'SLEEP',
+    'SPLIT',
+    'SOUNDEX',
+    'SOUNDS LIKE',
+    'SOURCE_POS_WAIT',
+    'SPACE',
+    'SQRT',
+    'STDDEV',
+    'STDDEV_POP',
+    'STDDEV_SAMP',
+    'STR_TO_DATE',
+    'SUBDATE',
+    'SUBSTR',
+    'SUBSTRING',
+    'SUBSTRING_INDEX',
+    'SUM',
+    'SYS_GUID',
+    'TAN',
+    'TIME',
+    'TIMEDIFF',
+    'TIME_BUCKET',
+    'TIME_FORMAT',
+    'TIMESTAMP',
+    'TIMESTAMPADD',
+    'TIMESTAMPDIFF',
+    'TIME_TO_SEC',
+    'TO_BASE64',
+    'TO_CHAR',
+    'TO_DAYS',
+    'TO_JSON',
+    'TO_NUMBER',
+    'TO_SECONDS',
+    'TO_TIMESTAMP',
+    'TRIM',
+    'TRUNC',
+    'TRUNCATE',
+    'UCASE',
+    'UNHEX',
+    'UNIX_TIMESTAMP',
+    'UPDATEXML',
+    'UPPER',
+    'USER',
+    'UTC_DATE',
+    'UTC_TIME',
+    'UTC_TIMESTAMP',
+    'UUID',
+    'VALUES',
+    'VARIANCE',
+    'VAR_POP',
+    'VAR_SAMP',
+    'VECTOR_SUB',
+    'VERSION',
+    'WEEK',
+    'WEEKDAY',
+    'WEEKOFYEAR',
+    'YEAR',
+    // Data types with parameters
+    // https://dev.mysql.com/doc/refman/8.0/en/data-types.html
```

Comment:
So, is the following list a list of MySQL data types or a list of SingleStoreDB data types?

Reply:
Yes, according to the reserved keywords: https://docs.singlestore.com/managed-service/en/reference/sql-reference/restricted-keywords/list-of-restricted-keywords.html

Reply:
So, why do we link to the MySQL docs here? Instead of e.g. https://docs.singlestore.com/managed-service/en/reference/sql-reference/data-types.html